lib/tests/formulae.rb

```diff
@@ -222,11 +222,17 @@ def bottled?(formula, no_older_versions:)
       end
 
       def build_bottle?(formula, args:)
-        all_deps_bottled = formula.deps.all? do |dep|
+        # Require build and runtime dependencies to be bottled on the current OS.
+        return false unless formula.deps.reject(&:test?).all? do |dep|
           bottled?(dep.to_formula, no_older_versions: true)
```

Comment:
Can reduce the duplication with

```suggestion
        return false unless formula.deps.all? do |dep|
          bottled?(dep.to_formula, no_older_versions: !dep.test?)
```